src/jsparse.c

```diff
@@ -1123,6 +1150,10 @@ NO_INLINE JsVar *jspeFactorMember(JsVar *a, JsVar **parentResult) {
         jsvUnLock(aVar);
       }
       jsvUnLock(index);
+
+#ifndef ESPR_NO_OPTIONAL_CHAINING
+      JSP_RESTORE_EXECUTE();
```

Comment:
I'm afraid this breaks stuff.  `undefined[1]` should cause an exception (like 1143) but JSP_RESTORE_EXECUTE removes the error flag, so it just quietly completes. I can imagine there might be other code you could come up with (not just errors, but return/break/etc) too